Seems like some worker function in the wifi code in 2.6.36 hits this bug_on,
which it shouldn't. It's hard to know if this is caused by BFS per se, or is
in iwlagn itself. It is, however, harmless to wake up a task on another RQ
with BFS, so just disable this BUG_ON for now.

-ck

---
 kernel/sched_bfs.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-2.6.36-bfs/kernel/sched_bfs.c
===================================================================
--- linux-2.6.36-bfs.orig/kernel/sched_bfs.c	2010-10-24 21:30:39.164151587 +1100
+++ linux-2.6.36-bfs/kernel/sched_bfs.c	2010-10-24 21:30:54.851780968 +1100
@@ -1460,7 +1460,6 @@ static void try_to_wake_up_local(struct
 	struct rq *rq = task_rq(p);
 	bool success = false;
 
-	BUG_ON(rq != this_rq());
 	BUG_ON(p == current);
 	lockdep_assert_held(&grq.lock);
 
